cartographer/mapping/internal/2d/pose_graph/optimization_problem_2d.cc

```diff
@@ -273,12 +273,28 @@ void OptimizationProblem2D::Solve(
         continue;
       }
 
-      const transform::Rigid3d relative_pose =
-          ComputeRelativePose(trajectory_id, first_node_data, second_node_data);
+      // Add a relative pose constraint based on the odometry (if available).
+      std::unique_ptr<transform::Rigid3d> relative_odometry =
+          CalculateOdometryBetweenNodes(trajectory_id, first_node_data,
+                                        second_node_data);
+      if (relative_odometry != nullptr) {
+        problem.AddResidualBlock(
+            SpaCostFunction2D::CreateAutoDiffCostFunction(Constraint::Pose{
+                (*relative_odometry), options_.odometry_translation_weight(),
```

Comment:
are these brackets necessary?

Reply:
Removing. I keep the ones in the multiplications though, they add better readability.